### PR TITLE
Use only 1 type for the User object

### PR DESCRIPTION
**Pull request by samerbuna**
Using multiple types mess with client caching

This PR is a requirement for https://github.com/GaloyMoney/galoy-mobile/pull/283

### CODE REVIEW

File: src/graphql/root/mutation/user-contact-update-alias.ts [hunk 1]
```diff
@@ -27,19 +27,19 @@ const USerContactUpdateAliasMutation = GT.Field({
...
-    if (user instanceof Error) {
-      return { errors: [{ message: user.message }] }
+    if (contact instanceof Error) {
+      return { errors: [{ message: contact.message }] }
     }
 
     return {
       errors: [],
-      user,
+      contact,
```
Comment (vindard) [inline src/graphql/root/mutation/user-contact-update-alias.ts:42]: `SuccessPayload` type has `success` key here instead I believe

File: src/graphql/types/payload/user-contact-update-alias.ts [hunk 2]
```diff
@@ -1,15 +1,16 @@
 import { GT } from "@graphql/index"
+
 import IError from "../abstract/error"
-import { UserDetails } from "../object/user"
+import UserContact from "../object/wallet-contact"
 
 const UserContactUpdateAliasPayload = new GT.Object({
```
Comment (vindard) [inline src/graphql/types/payload/user-contact-update-alias.ts:6]: Is this still needed now given that we're using `SuccessPayload`?

**Review by vindard — COMMENTED** [review]
Looks like a straight switchover to using our `User` type everywhere instead of having partial `User` types. I don't know enough about the tradeoffs of that with graphql so this looks ok code-structure wise.

Just left two small comments that may lead to potential changes.

File: src/graphql/root/mutation/user-contact-update-alias.ts (hunk 1 above)
Comment (samerbuna) [reply src/graphql/root/mutation/user-contact-update-alias.ts:42]: Rebased to fix this and merge with new description fields.

I made the payload here to be a contact user.

**Review by bodymindarts — DISMISSED** [review]

**Review by vindard — DISMISSED** [review]

**Review by vindard — APPROVED** [review]

**Review by bodymindarts — APPROVED** [review]

**Pull request merged**